relay-general/src/protocol/contexts/browser.rs

```diff
@@ -21,3 +24,229 @@ impl BrowserContext {
         "browser"
     }
 }
+
+impl FromUserAgentInfo for BrowserContext {
+    fn from_client_hints(client_hints: &user_agent::ClientHints) -> Option<Self> {
+        let browser = Browser::from_ua_client_hint(client_hints.sec_ch_ua?)?;
+        let version = client_hints.sec_ch_ua_full_version?.to_owned();
+
+        Some(Self {
+            name: Annotated::new(browser.to_string()),
+            version: Annotated::new(version),
+            ..Default::default()
+        })
+    }
+
+    fn from_user_agent(user_agent: &str) -> Option<Self> {
+        let browser = user_agent::parse_user_agent(user_agent);
+
+        if !store::user_agent::is_known(&browser.family) {
+            return None;
+        }
+
+        Some(Self {
+            name: Annotated::from(browser.family),
+            version: Annotated::from(store::user_agent::get_version(
+                &browser.major,
+                &browser.minor,
+                &browser.patch,
+            )),
+            ..BrowserContext::default()
+        })
+    }
+}
+
+#[derive(Debug, PartialEq)]
+enum Browser {
+    Chrome,
+    Edge,
+    Safari,
+    Brave,
+    Opera,
+    Firefox,
+    Other(String),
+}
+
+impl std::convert::From<String> for Browser {
+    fn from(s: String) -> Self {
+        match s.as_str() {
+            "Google Chrome" => Browser::Chrome,
+            "Microsoft Edge" => Browser::Edge,
+            "Firefox" => Browser::Firefox,
+            "Safari" => Browser::Safari,
+            "Brave" => Browser::Brave,
+            "Opera" => Browser::Opera,
+            _ => Browser::Other(s.to_string()),
+        }
+    }
+}
+
+impl std::string::ToString for Browser {
+    fn to_string(&self) -> String {
+        match self {
+            Browser::Chrome => "Google Chrome".to_string(),
+            Browser::Edge => "Microsoft Edge".to_string(),
+            Browser::Firefox => "Firefox".to_string(),
+            Browser::Safari => "Safari".to_string(),
+            Browser::Brave => "Brave".to_string(),
+            Browser::Opera => "Opera".to_string(),
```

Comment:
There's a few more browsers that already support `sec-ch-ua`, let's add them to this list. See https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Sec-CH-UA#browser_compatibility.